src/navigation/pathcurve.cpp

```diff
@@ -84,49 +95,89 @@ void PathCurve::initializeParameterData() {
     }
     _totalLength = _lengthSums.back();
 
+    if (_totalLength < LengthEpsilon) {
+        throw TooShortPathError("Path too short!");
+    }
+
     // Compute a map of arc lengths s and curve parameters u, for reparameterization
-    _parameterSamples.reserve(NrSamplesPerSegment * _nSegments + 1);
-    const double uStep = 1.0 / (_nSegments * NrSamplesPerSegment);
+    constexpr const int steps = 100;
+    const double uStep = 1.0 / static_cast<double>(steps);
+    _parameterSamples.reserve(steps * _nSegments + 1);
+
+    bool problematic = false;
+
     for (unsigned int i = 0; i < _nSegments; i++) {
         double uStart = _curveParameterSteps[i];
         double sStart = _lengthSums[i];
-        for (int j = 0; j < NrSamplesPerSegment; ++j) {
+        _parameterSamples.push_back({ uStart, sStart });
+        // Intermediate sampels
+        for (int j = 1; j < steps; ++j) {
             double u = uStart + j * uStep;
             double s = sStart + arcLength(uStart, u);
+            // Ignore samples that are indistinguishable due to precision limitations
+            if (std::abs(s - _parameterSamples.back().s) < LengthEpsilon) {
+                problematic = true;
+                continue;
+            }
             _parameterSamples.push_back({ u, s });
         }
     }
-    _parameterSamples.push_back({ 1.0, _totalLength });
+
+    if (problematic) {
```

Comment:
Why the extra boolean variable?  Shouldn't you be able to just throw the exception inside the for loop directly?

Reply:
True. With the current implementation, it can be done directly in the loop. I'll update so that it does

It was nice to keep the entire list of samples around for debugging purposes, but there really is no reason to keep it now that linear paths are used for this scenario